cmd/flipt-edge/main.go

```diff
@@ -0,0 +1,431 @@
+		// default to info level
+		loggerConfig.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
+	}
+
+	if cfg.Log.Encoding > config.LogEncodingConsole {
+		loggerConfig.Encoding = cfg.Log.Encoding.String()
+
+		// don't encode with colors if not using console log output
+		loggerConfig.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	}
+
+	logger := zap.Must(loggerConfig.Build())
+
+	// print out any warnings from config parsing
+	for _, warning := range res.Warnings {
+		logger.Warn("configuration warning", zap.String("message", warning))
+	}
+
+	if found {
+		logger.Debug("configuration source", zap.String("path", path))
+	}
+
+	return logger, cfg, nil
+}
+
+const (
+	dntVar = "DO_NOT_TRACK"
+	ciVar  = "CI"
+)
+
+func isSet(env string) bool {
+	return os.Getenv(env) == "true" || os.Getenv(env) == "1"
+}
+
+func run(ctx context.Context, logger *zap.Logger, cfg *config.Config) error {
+	isConsole := cfg.Log.Encoding == config.LogEncodingConsole
+
+	if isConsole {
+		color.Cyan("%s\n", banner)
+	} else {
+		logger.Info("flipt starting", zap.String("version", version), zap.String("commit", commit), zap.String("date", date), zap.String("go_version", goVersion))
+	}
+
+	var (
+		isRelease   = release.Is(version)
+		releaseInfo release.Info
+		err         error
+	)
+
+	if cfg.Meta.CheckForUpdates && isRelease {
+		logger.Debug("checking for updates")
+
+		releaseInfo, err = release.Check(ctx, version)
+		if err != nil {
+			logger.Warn("checking for updates", zap.Error(err))
+		}
+
+		logger.Debug("version info", zap.String("current_version", releaseInfo.CurrentVersion), zap.String("latest_version", releaseInfo.LatestVersion))
+
+		if isConsole {
+			if releaseInfo.UpdateAvailable {
+				color.Yellow("A newer version of Flipt exists at %s, \nplease consider updating to the latest version.", releaseInfo.LatestVersionURL)
+			} else {
+				color.Green("You are currently running the latest version of Flipt [%s]!", releaseInfo.CurrentVersion)
+			}
+		} else {
+			if releaseInfo.UpdateAvailable {
+				logger.Info("newer version available", zap.String("version", releaseInfo.LatestVersion), zap.String("url", releaseInfo.LatestVersionURL))
+			} else {
+				logger.Info("running latest version", zap.String("version", releaseInfo.CurrentVersion))
+			}
+		}
+	}
+
+	// see: https://consoledonottrack.com/
+	if isSet(dntVar) && cfg.Meta.TelemetryEnabled {
+		logger.Debug("DO_NOT_TRACK environment variable set, disabling telemetry")
+		cfg.Meta.TelemetryEnabled = false
+	}
+
+	if isSet(ciVar) && cfg.Meta.TelemetryEnabled {
+		logger.Debug("CI detected, disabling telemetry")
+		cfg.Meta.TelemetryEnabled = false
+	}
+
+	if !isRelease && cfg.Meta.TelemetryEnabled {
+		logger.Debug("not a release version, disabling telemetry")
+		cfg.Meta.TelemetryEnabled = false
+	}
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	if err := initMetaStateDir(cfg); err != nil {
```

Comment:
also will prob want to adjust the name of this state file incase someone runs Flipt Edge and Flipt on the same machine (at different times)

Reply:
Good shout!

Reply:
After looking at this one too. I think actually leaving this path as-is, is both convient from a complexity standpoint, as well as being reasonably safe to do.

I think for 1, people will likely not run them on the same machine.
At-least, they will likely run these in a containerized environment. In which case, they're already isolated.

Secondly, edge consumes a subset of the full config. So it may just work.
Also, we never overwrite this config, we just read it. So I don't think we have a chance of corrupting anything.

If anything, it may just be useful that you can switch between the two without having to do much.